app/js/elements/fxos-inspector.js

```diff
@@ -0,0 +1,288 @@
+(function(window) {
+'use strict';
+
+const HTML_ESCAPE_CHARS = {
+  '&': '&amp;',
+  '<': '&lt;',
+  '>': '&gt;',
+  '"': '&quot;',
+  '\'': '&#39;',
+  '/': '&#x2F;'
+};
+
+const CONSTANT_PROPERTY_REGEX = /^[A-Z_]+$/;
+
+var proto = Object.create(HTMLElement.prototype);
+
+var template =
+`<style scoped>
+  [flex] {
+    display: flex;
+    justify-content: space-between;
+    padding: 0 5px;
+  }
+  [flex] > * {
+    flex: 1 1 auto;
+    margin-right: 5px;
+    margin-left: 5px;
+  }
+  .container {
+    box-sizing: border-box;
+    position: relative;
+    width: 100%;
+    height: 100%;
+    overflow-x: hidden;
+    overflow-y: auto;
+  }
+  gaia-list a {
+    text-decoration: none;
+  }
+  gaia-list a[hidden] {
+    display: none;
+  }
+  gaia-list h3 {
+    max-width: calc(100% - 2em);
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+  gaia-list .value {
+    color: #00aac5;
+  }
+  fxos-code-editor {
+    display: block;
+    width: 100%;
+    height: calc(100% - 100px);
+  }
+</style>
+<div class="container"></div>`;
+
+proto.createdCallback = function() {
+  this.shadow = this.createShadowRoot();
+  this.shadow.innerHTML = template;
+
+  this.container = this.shadow.querySelector('.container');
+
+  this.keyPath = [];
+
+  this.container.addEventListener('click', (evt) => {
+    var target = evt.target.closest('a');
+    if (!target) {
+      target = evt.target.closest('gaia-button');
+      if (!target) {
+        return;
+      }
+
+      // Handle case where <gaia-button> was clicked.
+      evt.preventDefault();
+
+      switch (target.dataset.action) {
+        case 'cancel':
+          this.keyPath.pop();
+          break;
+        case 'save':
+          (() => {
+            var value = this.container.querySelector('fxos-code-editor').value;
+            var target = this.rootTarget;
+
+            this.keyPath.forEach((key, index) => {
```

Comment:
It took me a minute to understand what this is doing. A comment would be helpful.